src/main/java/software/amazon/cloudwatchlogs/emf/model/MetricDirective.java

```diff
@@ -89,6 +89,16 @@ void setDimensions(List<DimensionSet> dimensionSets) {
         dimensions = new ArrayList<>(dimensionSets);
     }
 
+    /**
+     * Clear existing custom dimensions.
+     *
+     * @param preserveDefault indicates whether default dimensions should be used
+     */
+    void resetDimensions(boolean preserveDefault) {
+        shouldUseDefaultDimension = preserveDefault;
+        dimensions = Collections.synchronizedList(new ArrayList<>());
```

Comment:
I understand that you will make it synchronized in following cr but for integrity and consistency of this cr let's use ArrayList

Reply:
Solved.